ios/Classes/SwiftFlutterWeatherSensorPlugin.swift

```diff
@@ -1,14 +1,76 @@
 import Flutter
 import UIKit
+import Combine
 
-public class SwiftFlutterWeatherSensorPlugin: NSObject, FlutterPlugin {
+@available(iOS 13.0, *)
+public class SwiftFlutterWeatherSensorPlugin: NSObject, FlutterPlugin, FlutterStreamHandler {
+  private var eventSink: FlutterEventSink? = nil
+  private var sinkSubscription: AnyCancellable?
+    
   public static func register(with registrar: FlutterPluginRegistrar) {
-    let channel = FlutterMethodChannel(name: "flutter_weather_sensor_plugin", binaryMessenger: registrar.messenger())
+    let methodChannel = FlutterMethodChannel(name: "flutter_weather_sensor_plugin", binaryMessenger: registrar.messenger())
+    let eventChannel = FlutterEventChannel(name: "flutter_weather_sensor_plugin/readings", binaryMessenger: registrar.messenger())
+      
     let instance = SwiftFlutterWeatherSensorPlugin()
-    registrar.addMethodCallDelegate(instance, channel: channel)
+    registrar.addMethodCallDelegate(instance, channel: methodChannel)
+    eventChannel.setStreamHandler(instance)
   }
 
   public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
-    result("iOS " + UIDevice.current.systemVersion)
+    if (call.method == "startSensorReadings") {
+      startSensorReadings()
+      result("success")
+    } else if (call.method == "stopSensorReadings") {
+      stopSensorReadings()
+      result("success")
+    } else if (call.method == "set") {
+      if let args = call.arguments as? Dictionary<String, Any>,
+        let interval = args["readingInterval"] as? NSInteger {
+          setInterval(interval)
+        } else {
+          result(FlutterError.init(code:"invalidArgs", message: "Reading interval must be an integer.", details: "Expected 1 Int arg."))
+        }
+      result("success")
+    } else if (call.method == "collectReadings") {
+      collectReadings()
+      result("success")
+    } else {
+      result(FlutterMethodNotImplemented)
+      return
+    }
   }
+    
+  public func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
+    eventSink = events
+    return nil
+  }
+    
+  public func onCancel(withArguments arguments: Any?) -> FlutterError? {
+    eventSink = nil
+    return nil
+  }
+    
+  private func collectReadings() {
+    sinkSubscription = WeatherSensorService.shared.reader.sensorReadingsPublisher.sink{ value in
+      if (self.eventSink != nil) {
+        self.eventSink!("\(value)")
+      }
+    }
+  }
```

Comment:
The code above could lead to a memory leak. If self is deallocated (and it might never get deallocated in this particular case, but neglecting handling it is not great), then `WeatherSensorService.shared` would still hold a reference to this class, and then this class would not be able to be deallocated. At least I think I have that right...

The easy way to get around that and what is almost always done is to create a weak reference to self in the closure, like this:
```
private func collectReadings() {
    sinkSubscription = WeatherSensorService.shared.reader.sensorReadingsPublisher.sink{ [weak self] value in
      if (self?.eventSink != nil) {
        self?.eventSink!("\(value)")
      }
    }
  }
```

Further, you could consolidate that code a little (and note the `()` are not needed, and the use of the force-unwrap `!` is sort of bad form and should be used very sparingly). If `self` or `eventSink` is nil, this will amount to a no-op:
```
private func collectReadings() {
    sinkSubscription = WeatherSensorService.shared.reader.sensorReadingsPublisher.sink{ [weak self] value in
      self?.eventSink?(value)
    }
  }
```